pkg/persistence/object_store.go

```diff
@@ -225,50 +227,51 @@ func (s *objectBackupStore) ListBackups() ([]string, error) {
 }
 
 func (s *objectBackupStore) PutBackup(info BackupInfo) error {
-	if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getBackupLogKey(info.Name), info.Log); err != nil {
-		// Uploading the log file is best-effort; if it fails, we log the error but it doesn't impact the
-		// backup's status.
-		s.logger.WithError(err).WithField("backup", info.Name).Error("Error uploading log file")
-	}
-
-	if info.Metadata == nil {
-		// If we don't have metadata, something failed, and there's no point in continuing. An object
-		// storage bucket that is missing the metadata file can't be restored, nor can its logs be
-		// viewed.
-		return nil
+	if info.Log != nil {
+		if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getBackupLogKey(info.Name), info.Log); err != nil {
+			// Uploading the log file is best-effort; if it fails, we log the error but it doesn't impact the
+			// backup's status.
+			s.logger.WithError(err).WithField("backup", info.Name).Error("Error uploading log file")
+		}
 	}
 
-	if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getBackupMetadataKey(info.Name), info.Metadata); err != nil {
-		// failure to upload metadata file is a hard-stop
-		return err
+	if info.Metadata != nil {
```

Comment:
I'm against this check, but could you remind me in what case the `Metadata`, `Content`, `Log` or `reader` is nil?
If it should not be nil, like mentioned in the old comment
```
there's no point in continuing
```
Maybe we should stop and fail early?

Reply:
I split the backup upload into the object store into two pieces for the upload progress monitoring.  It used to all go as one unit.  The Metadata goes into the velero-backup.json file.  That controls whether the backup is considered usable.  reader is going to be for the things that are in the backupObjs map and those are all optional and any of them can be nil.  I think the extra check I added for nil there is redundant, I'll take that out.

We need to upload things like the backup tar file itself (Content), the log files (Log) and possibly things that were in the backupObjs map  when we finish phase 1 of the backup, but without the ItemSnapshots (which would be in the backupObjs map).  After we finish phase 1, we will go into the monitoring phase and another backup may begin.  The in-memory structures may disappear if we take a restart so the upload after the final phase just consists of the Metadata and ItemSnapshots and everything else will be nil.  It seemed reasonable to leave everything together in PutBackup as the checks were mainly for panic situations.

Reply:
OK, done